flang/runtime/pointer.cpp

```diff
@@ -124,6 +124,23 @@ void RTDEF(PointerAssociateRemapping)(Descriptor &pointer,
   }
 }
 
+RT_API_ATTRS void *AllocateValidatedPointerPayload(std::size_t byteSize) {
+  // Add space for a footer to validate during deallocation.
+  constexpr std::size_t align{sizeof(std::uintptr_t)};
+  byteSize = ((byteSize / align) + 1) * align;
```

Comment:
```suggestion
  byteSize = ((byteSize + align - 1) / align) * align;
```

Reply:
If `byteSize` is initially zero, I want the result to be `align`, not zero.

Reply:
Can you explain why?  If `byteSize` is initially zero, the total allocation will end up being `2 * align` - isn't `align` bytes allocation enough?

Reply:
`((0/align)+1)*align` -> `(0+1)*align` -> `1*align` -> `align`, yes?

Reply:
Yes, `byteSize` is equal to `align`, and then `total` (below) is equal to `align + sizeof(std::uintptr_t)` - this is how much memory is being malloc'ed.  I am asking why we need to malloc more than `align` bytes of memory.

Reply:
We don't.  Will fix.  I was misunderstanding your comment.

Reply:
https://github.com/llvm/llvm-project/pull/96663